sycl/source/handler.cpp

```diff
@@ -1057,6 +1036,8 @@ void handler::ext_oneapi_graph(
     auto SubgraphEvent = std::make_shared<event_impl>();
     ParentGraph->add_event_for_node(SubgraphEvent, MSubgraphNode);
   } else {
+    MIsFinalized = true;
+    // TODO: This could probably be moved to handler::finalize()
```

Comment:
How much work would it be to do this as part of this PR?

Reply:
Turned out to be mostly trivial (just needs a new handler member) so I've done it now!